lib/models/device.ts

```diff
@@ -3436,6 +3440,140 @@ const getDeviceModel = function (
 				});
 			},
 		}),
+
+		/**
+		 * @namespace balena.models.device.history
+		 * @memberof balena.models.device
+		 */
+		history: {
+			/**
+			 * @summary Get all history entries for a device
+			 * @name getAllByDevice
+			 * @public
+			 * @function
+			 * @memberof balena.models.device.history
+			 *
+			 * @param {String|Number} uuidOrId - device uuid (32 / 62 digits string) or id (number)
+			 * @param {Object} [options] - options
+			 * @param {Date} [options.fromDate=subDays(new Date(), 7)] - history entries older or equal to this date - default now() - 7 days
+			 * @param {Date} [options.toDate] - history entries younger or equal to this date
+			 * @fulfil {Object[]} - device history
+			 * @returns {Promise}
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByDevice('7cf02a687b74206f92cb455969cf8e98').then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByDevice(999999).then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 *
+			 * @example
+			 * // get all device history entries between now - 20 days and now - 10 days
+			 * balena.models.device.history.getAllByDevice(999999, { fromDate: subDays(new Date(), 20), toDate: subDays(new Date(), 10)})
+			 */
+			async getAllByDevice(
+				uuidOrId: string | number,
+				{
+					fromDate,
+					toDate,
+					...options
+				}: PineOptions<DeviceHistory> & { fromDate?: Date; toDate?: Date } = {
+					fromDate: subDays(new Date(), defaultDaysOfRequestedHistory),
+				},
+			): Promise<DeviceHistory[]> {
+				let $filter = {
```

Comment:
Can you try doing v :
```suggestion
				const $filter: PineFilter<DeviceHistory> = {
```
and then see whether TS complains if you do `$filter.tracks__device = uuidOrId`.


If that doesn't work, then let's just add the type here anyway even w/ a `let`:
```
let $filter: PineFilter<DeviceHistory>
```

Reply:
Typescript unfortunately complains and the $filter comes without member knowledge inthe IDE.